apps/desktop-wallet/src/components/AssetLogo.tsx

```diff
@@ -16,40 +16,49 @@ You should have received a copy of the GNU Lesser General Public License
 along with the library. If not, see <http://www.gnu.org/licenses/>.
 */
 
-import { FungibleToken, NFT } from '@alephium/shared'
 import { HelpCircle } from 'lucide-react'
+import { memo } from 'react'
 import ReactPlayer from 'react-player'
 import styled from 'styled-components'
 
+import useToken, { isFT, isListedFT, isNFT } from '@/api/apiDataHooks/useToken'
+import { TokenId } from '@/types/tokens'
+
 interface AssetLogoProps {
-  assetImageUrl: FungibleToken['logoURI'] | NFT['image']
+  tokenId: TokenId
   size: number
-  assetName?: FungibleToken['name']
-  isNft?: boolean
   className?: string
 }
 
-const AssetLogo = ({ assetImageUrl, size, assetName, className }: AssetLogoProps) => (
-  <div className={className}>
-    {assetImageUrl?.endsWith('.mp4') ? (
-      <ReactPlayer url={assetImageUrl} muted width={size} height={size} />
-    ) : assetImageUrl ? (
-      <LogoImage src={assetImageUrl} />
-    ) : assetName ? (
-      <Initials size={size}>{assetName.slice(0, 2)}</Initials>
-    ) : (
-      <HelpCircle size={size} />
-    )}
-  </div>
-)
+const AssetLogo = memo(({ tokenId, size, className }: AssetLogoProps) => {
+  const { data: token } = useToken(tokenId)
+  const image = isListedFT(token) ? token.logoURI : isNFT(token) ? token.image : undefined
+  const name = isFT(token) || isNFT(token) ? token.name : undefined
+
+  return (
+    <AssetLogoStyled className={className} size={size} isSquare={isNFT(token)}>
+      {image?.endsWith('.mp4') ? (
```

Comment:
This reminds me that we should port the more generic and efficient solution from the explorer, the one creating thumbnails and compatible with more video formats.

https://github.com/alephium/alephium-frontend/blob/next/apps/explorer/src/components/VideoThumbnail.tsx

Reply:
Wanna create an issue to not forget about it?